Source/OxyPlot.Wpf/CanvasRenderContext.cs

```diff
@@ -789,46 +956,25 @@ private void ApplyToolTip(FrameworkElement element)
             double[] dashArray,
             LineJoin lineJoin)
         {
-            StreamGeometry streamGeometry = null;
-            StreamGeometryContext streamGeometryContext = null;
+            if (points.Count < 2)
+                return;
+
+            var path = this.CreateAndAdd<Path>();
+            this.SetStroke(path, stroke, thickness, edgeRenderingMode, lineJoin, dashArray, 0);
 
-            int count = 0;
             var actualStrokeThickness = this.GetActualStrokeThickness(thickness, edgeRenderingMode);
+            var actualPoints = this.GetActualPointsForStreamGeometry(points, actualStrokeThickness, edgeRenderingMode);
 
-            for (int i = 0; i + 1 < points.Count; i += 2)
+            var streamGeometry = new StreamGeometry();
+            using (var streamGeometryContext = streamGeometry.Open())
             {
-                if (streamGeometry == null)
+                streamGeometryContext.BeginFigure(actualPoints.Item1, false, false);
+                for (int i = 0; i < actualPoints.Item2.Count; i++)
                 {
-                    streamGeometry = new StreamGeometry();
-                    streamGeometryContext = streamGeometry.Open();
-                }
-
-                var actualPoints = this.GetActualPoints(new[] { points[i], points[i + 1] }, actualStrokeThickness, edgeRenderingMode).ToList();
-
-                streamGeometryContext.BeginFigure(actualPoints[0], false, false);
-                streamGeometryContext.LineTo(actualPoints[1], true, false);
-
-                count++;
-
-                // Must limit the number of figures, otherwise drawing errors...
```

Comment:
Are we sure we don't create any regressions by removing this?
To be honest, I have no idea which issue this code originally solved, but I guess it was here for some reason.
For what it's worth, on my machine I don't get any drawing errors with or without this. If we can pin this to one specific issue in WPF rendering that is solved by now, I guess it is safe to remove this, otherwise I'm not sure...

Reply:
I'm pretty sure. This value is here to limit the number of figures per geometry, but this changes it to having a single geometry with a single figure in it, so this limitation doesn't apply anymore.

Reply:
Ah I see, thanks for the clarification!